osquery/tables/system/darwin/firewall.cpp

```diff
@@ -43,6 +44,61 @@ const std::map<std::string, std::string> kTopLevelStringKeys{
     {"version", "version"},
 };
 
+Status parseApplicationAliasData(const std::string& data, std::string& result) {
+
+  std::string decoded_data = base64Decode(data);
+  if (decoded_data.length() == 0) {
+    return Status(1, "Failed to base64 decode data");
+  }
+
+  CFDataRef resourceData = CFDataCreate(nullptr, (const UInt8*)decoded_data.data(), decoded_data.length());
+  if (resourceData == nullptr) {
+    return Status(1, "Failed to allocate resource data");
+  }
+
+  CFErrorRef errorString;
+  CFDataRef alias = (CFDataRef)CFPropertyListCreateWithData(kCFAllocatorDefault, resourceData, kCFPropertyListImmutable, nullptr, &errorString);
+  if (alias == nullptr) {
+    CFRelease(resourceData);
+    CFRelease(errorString);
```

Comment:
I'd recommend placing a check for `errorString != nullptr`, regardless of the return from `CFPropertyListCreateWithData`, and freeing. But if you don't care about the output can you provide `nullptr` as the 5th param too?